### PR TITLE
Add more inspect checks for use of deprecated components

### PR DESCRIPTION
**Pull request by K-ballo**

### CODE REVIEW

File: tools/inspect/deprecated_include_check.cpp [hunk 1]
```diff
@@ -29,7 +29,9 @@ namespace boost
       { "boost/unordered_set\\.hpp", "unordered_set" },
       { "boost/utility/enable_if\\.hpp", "type_traits" },
       { "boost/detail/atomic_count\\.hpp", "hpx/util/atomic_count.hpp" },
-      { "boost/function\\.hpp", "hpx/util/function.hpp" },
```
Comment (hkaiser) [inline tools/inspect/deprecated_include_check.cpp]: Why have you removed the warning when boost/function.hpp is included?

Comment (hkaiser) [inline tools/inspect/deprecated_include_check.cpp]: I take that back, it was re-added by a later commit.

File: src/runtime/agas/big_boot_barrier.cpp [hunk 2]
```diff
@@ -7,6 +7,8 @@
 //  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
 ////////////////////////////////////////////////////////////////////////////////
 
+// hpxinspect:nodeprecatedname:boost::unique_lock
```
Comment (hkaiser) [inline src/runtime/agas/big_boot_barrier.cpp]: Can't we use `std::unique_lock` with `boost::mutex`?

File: tools/inspect/deprecated_include_check.cpp (hunk 1 above)
Comment (K-ballo) [inline tools/inspect/deprecated_include_check.cpp]: I didn't even noticed I had removed it, and later added a duplicated check. Will rewrite the fix.

File: src/runtime/agas/big_boot_barrier.cpp (hunk 2 above)
Comment (K-ballo) [inline src/runtime/agas/big_boot_barrier.cpp]: With `boost::mutex` yes, with `boost::condition_variable` no.

**Pull request merged**